tests/ord-aggregator/tests/main_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tests
+
+import (
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	c "github.com/robfig/cron/v3"
+	"github.com/vrischmann/envconfig"
+)
+
+type config struct {
+	DefaultTenant                 string
+	DirectorURL                   string
+	ORDServiceURL                 string
+	AggregatorSchedule            string
+	ExternalServicesMockBaseURL   string
+	ORDServiceDefaultResponseType string
+}
+
+var testConfig config
+
+func TestMain(m *testing.M) {
+	err := envconfig.Init(&testConfig)
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "while initializing envconfig"))
+	}
+	exitVal := m.Run()
+	os.Exit(exitVal)
+
+}
+
+func parseCronTime(cronTime string) (time.Duration, error) {
+	parser := c.NewParser(c.Minute | c.Hour | c.Dom | c.Month | c.Dow)
+	scheduleTime, err := parser.Parse(cronTime)
+	if err != nil {
+		return 0, errors.New("error while parsing cron time")
+	}
+
+	// This is the starting time that will be subtracted from the next activation cron time below. This way the cron time duration can be estimated.
+	year, month, day := time.Now().Date()
+	startingTime := time.Date(year, month, day, 0, 0, 0, 0, time.Now().Location())
+
+	nextTime := scheduleTime.Next(startingTime)
+	cronTimeDuration := nextTime.Sub(startingTime)
```

Comment:
can't this be simplified like this:

```go
now := time.Now()
next := scheduleTime.Next(now)
next.Sub(now)
```

Reply:
That was my initial approach but it is incorrect based on how `scheduleTime.Next()` works. In the current setup, the cron time that we pass has value of 5mins (execute every 5mins). By design, that means execution in: xx:05, xx:10, xx:15, xx:20 etc.

`scheduleTime.Next(<time>)` gives you the next scheduled execution after the given `<time>`. So, if we pass `time.Now()`  and then subtract it, the result will be how much time is left to the next scheduled execution (e.g: 
time.Now() is 10:12, `next := scheduleTime.Next(now); next.Sub(now)` will result in 3 mins: "10:15 - 10:12"). So, it is impossible to calculate how much is the cron time.

With the implemented approach, we set a “fixed” initial time (00:00), so that when we subtract it from the next scheduled cron time (00:05 in our case), the result will be the actual duration of the cron time. You can think of it as subtraction of zero.